tuxedo-core/aggregator/src/lib.rs

```diff
@@ -115,20 +115,39 @@ pub fn tuxedo_constraint_checker(attrs: TokenStream, body: TokenStream) -> Token
     let variants = variant_type_pairs.clone().map(|(v, _t)| v);
     let inner_types = variant_type_pairs.map(|(_v, t)| t);
 
-    let vis = ast.vis;
+    // Set up the names of the new associated types.
     let mut error_type_name = outer_type.to_string();
     error_type_name.push_str("Error");
     let error_type = Ident::new(&error_type_name, outer_type.span());
+
+    let mut accumulator_type_name = outer_type.to_string();
+    accumulator_type_name.push_str("Accumulator");
+    let accumulator_type = Ident::new(&accumulator_type_name, outer_type.span());
+
+    let mut accumulator_value_type_name = outer_type.to_string();
+    accumulator_value_type_name.push_str("AccumulatorValueType");
+    let accumulator_value_type = Ident::new(&accumulator_value_type_name, outer_type.span());
+
+    let vis = ast.vis;
     let inner_types = inner_types.clone();
+    let inner_types2 = inner_types.clone();
+    let inner_types3 = inner_types.clone();
+    let inner_types4 = inner_types.clone();
+    let inner_types5 = inner_types.clone();
     let variants2 = variants.clone();
+    let variants3 = variants.clone();
+    let variants4 = variants.clone();
+    let variants5 = variants.clone();
+    let variants6 = variants.clone();
```

Comment:
There's gotta be a better way to do this right? Does anyone know? I'm pretty much a macro newb. This crate is by far the most complex one I've ever written.